sdk/template/template/test/public/core-v2-test.spec.ts

```diff
@@ -0,0 +1,71 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { TableClient, TableEntity } from "@azure/data-tables";
+import { TestProxyHttpClient, recorderHttpPolicy } from "@azure-tools/test-recorder-new";
+import { isPlaybackMode, env } from "@azure-tools/test-recorder";
+import { Context } from "mocha";
+import { SanitizerOptions } from "@azure-tools/test-recorder-new";
+import { createSimpleEntity } from "../utils/utils";
+
+// A fake connection string which replaces the actual connection string in the recording.
+const fakeConnString =
+  "TableEndpoint=https://fakeaccountname.table.core.windows.net/;SharedAccessSignature=st=2021-08-03T08:52:15Z&spr=https&sig=fakesigval";
+const sanitizerOptions: SanitizerOptions = {
+  // We use a connection string sanitizer to remove the real connection string in the recording.
+  connectionStringSanitizers: [
+    {
+      actualConnString: env.TABLES_SAS_CONNECTION_STRING,
+      fakeConnString
+    }
+  ],
+  // The removeHeaderSanitizer can be used to remove certain headers before saving a recording.
+  removeHeaderSanitizer: { headersForRemoval: ["X-Content-Type-Options"] }
+};
+
+const recorderOptions = {
+  envSetupForPlayback: {
+    TABLES_SAS_CONNECTION_STRING: fakeConnString
+  },
+  sanitizerOptions
+};
+
+describe("Core V2 tests using unified recorder", () => {
+  let recorder: TestProxyHttpClient;
+
+  // Note the use of function() instead of an arrow function. This is important
+  // since we pass `this.currentTest` to the client.
+  beforeEach(async function(this: Context) {
+    // Create an instance of recorder. We pass the current test as required by the client;
+    // this value is used to determine the recording's location.
+    recorder = new TestProxyHttpClient(this.currentTest);
+
+    // Start the recorder using the options before each test.
+    await recorder.start(recorderOptions);
+  });
+
+  afterEach(async () => {
+    await recorder.stop();
+  });
+
+  it("data-tables create entity", async function() {
+    // Variables can be generated in record mode, and are stored with the recording to be reused
+    // in playback mode.
+    if (!isPlaybackMode()) {
+      recorder.variables["table-name"] = `table${Math.ceil(Math.random() * 1000 + 1000)}`;
+    }
+    const client = TableClient.fromConnectionString(
+      env.TABLES_SAS_CONNECTION_STRING,
+      recorder.variables["table-name"]
+    );
+
+    // To set up the recorder with the core v2 pipeline, we add a HTTP policy which
+    // redirects requests to use the proxy tool recorder.
+    client.pipeline.addPolicy(recorderHttpPolicy(recorder));
```

Comment:
I think it could be nicer to invert this call relationship:

`recorder.instrument(client)`

or something like that.

Reply:
Hmmm, thinking.

Reply:
I wonder if it would be possible to somewhat unify the core v1 and core v2 approaches in this way, e.g.

```ts
const client = recorder.instrument(MyClientClass, /* ...rest of args passed through to the client constructor... */);
```
And `recorder.instrument` would detect whether the passed client class is a core v1 or core v2 SDK and do the right thing. Just a random thought -- I suspect it's too much TypeScript voodoo and it definitely obfuscates what's going on a bit. Doubt it's worthwhile.

Reply:
Let's ignore core-v1 since we move to core-v2 for all the packages at some point.

Quick PR for the core-v2 https://github.com/Azure/azure-sdk-for-js/pull/19362 🙂